### ISSUE TITLE
Help needed?

### ISSUE THREAD

**Issue by yarikoptic** [warner/python-versioneer#211]
Versioneer got loved by many, thus of all **big thank you** @warner for it!

But is it the right assessment of the situation that its maintenance/development is somewhat "froze up"?

I know e.g. that some projects started to use custom forks (https://github.com/nipy/nibabel/blob/master/versioneer.py) and thus duplicating efforts etc. 

May be you @warner  would accept co-maintainership and/or even move it from under your personal wing under a dedicated https://github.com/python-versioneer organization ?  Not that it miraculously would resolve all the issues at once, but I think it might pave the way forward.

Thank you in advance for reply.

Ping @effigies @jwodder @mih

**Comment by bashtage** [comment]
While this is clearly the best path, the alternative would be to setup an org with an unofficial but hopefully widely adopted fork that could become a defacto successor.

**Comment by yarikoptic** [comment]
Yes @bashtage , that would be they path to proceed if no reply follows.

**Comment by bucknerns** [comment]
I agree an org where multiple participants pave the way forward would be better.
I decided to jump on this and created the org.

**Comment by bashtage** [comment]
> created the org.

link to repo?

**Comment by bucknerns** [comment]
I invited @yarikoptic and @bashtage as owners of the org. https://github.com/python-versioneer

**Comment by effigies** [comment]
Care to open up issues on that branch? IMO, a good first step would be to curate some of the issues from here.

In the meantime, I would say the most important questions are:

1) What versions of Python should the fork support?
2) Continue with vendoring, disable it, or make optional?

Also worth figuring out an overall scope to the fork, to help determine whether a request should be acted on. I think the legitimacy of the fork will be helped by a maintenance team that's on the same page.

Speaking of which, I'm happy to help with this effort, as well.

**Comment by bucknerns** [comment]
So currently we don't have access to the pypi project so pushing new versions of versioneer is not an option. I think we should split apart the versioneer.py file in the versioneer-518 repo remove python2 compatibility and support 3.6+. Also I would remove the vendoring and just use the pep 518 style.

**Comment by effigies** [comment]
IMO including PEP numbers is needless cognitive overhead. For users not really deep in the packaging weeds (which I don't think we should assume people wanting to use versioneer are), 518 isn't going to be very meaningful and invites versioneer-517 as a typosquat. Would you be open to forking to a new name?

A couple that come to mind that are not yet claimed on PyPI are:

* Postfixed
  * versionaut (a la astro-/cosmo-naut)
  * versionite (or -ate, or -ete, for a chemistry vibe)
* Prefixed
  * aversion
  * diversion
  * inversion
  * perversion

I'm a bit partial to diversion, as we're forking, but also happy to hear other suggestions.

> I think we should split apart the versioneer.py file

Agreed.

> remove python2 compatibility and support 3.6+.

Agreed. Python 3.5 is end-of-life at the end of this month. I would suggest following EOL pretty strictly and not advancing on it as some projects I'm involve in are doing.

> I would remove the vendoring and just use the pep 518 style.

I'm definitely up for this as the default, but `pyproject.toml` does break use cases like `pip install --user --editable` (see https://github.com/nipy/nibabel/issues/859#issuecomment-572238805). Might be worth seeing if a `--vendor` flag is too difficult to maintain?

**Comment by yarikoptic** [comment]
Thank you @bucknerns , so a fork it is :-/  I still hope for @warner to re-emerge and will try to figure out some direct communication channel.
I took advantage of the "Owner" status and invited @effigies as yet another Owner -- I would trust him more than myself with that role  ;)

**Comment by bashtage** [comment]
> * Postfixed
>   
>   * versionaut (a la astro-/cosmo-naut)
>   * versionite (or -ate, or -ete, for a chemistry vibe)
> * Prefixed
>   
>   * aversion
>   * diversion
>   * inversion
>   * perversion (I like it as per-version, but probably too creepy to be worth the pun)

I would prefer to keep the `eer` on the name, which would mean a prefix, or a post-fix.  If @warner is unable to particulate, we could consider initiating the process with Python.org to get a PyPI record owner changed. I believe this is quite a slow process, as it should be.

**Comment by bashtage** [comment]
Would it make sense to ask for PRs to be opened against python-versioneer/python-versioneer?

**Comment by effigies** [comment]
> * reversioneer

This would be my preference of this set. It's not obvious to me that we need to keep the `eer` to establish lineage (that's what I'm assuming this is about), but I'm happy to go that way if others disagree.

> Would it make sense to ask for existing PRs to be opened against python-versioneer/python-versioneer?

Should probably curate which ones first. The cleanup ones (#177, #190, #207) and #163 all look pretty uncontroversial. Others might need more thinking about. Particularly ones with merge conflicts, since it will take some work for submitters to update them, which would be wasted if they don't seem likely to be accepted.

**Comment by warner** [comment]
Hey folks.. I'm so sorry I didn't respond, yeah, life/work is busy these days. I'd be happy to transfer ownership over to the group. I've got some time this weekend, let me know how you want to move things. I could transfer the repo to the new org is that's the easiest approach. Thanks!

**Comment by bucknerns** [comment]
I invited @warner to the org to facilitate the transfer and I think it's in our common interest to keep the creator involved so we can ask questions we may have.

**Comment by bashtage** [comment]
This is great news, thanks @warner.  Would it be possible to add someone
else as an owner of the pypi record?

**Comment by yarikoptic** [comment]
@warner I am yarikoptic on pypi as well.

**Comment by bucknerns** [comment]
I am bucknerns on pypi.

**Comment by warner** [comment]
github transfer complete

**Comment by warner** [comment]
pypi invites sent

**Comment by effigies** [comment]
Thanks @warner!

I've gotten us set up on TravisCI.com. Currently failing tests, and #190 is no longer fixing them. My inclination is to drop Python < 3.6 before even trying to get the CI back in working order...

**Comment by bashtage** [comment]
Thanks for all of the help @warner . versioneer has been a fantastic module that has taken a lot of pain out of both testing and releasing and has really been a fantastic asset to the community.

**Comment by alimcmaster1** [comment]
Agree^ thanks all for transitioning this!

**Comment by effigies** [comment]
I've created a 0.19 milestone, and I'm going to start populating it with what I think are low-hanging fruit to finish off a quick release before there are any major refactorings.

I would suggest we target early October.

Also, as a heads up, I set the version to `0.19.dev0` to avoid confusion over 0.18 in people's built `versioneer.py` files when they're testing out changes since the release.

**Comment by effigies** [comment]
@yarikoptic @bashtage @bucknerns I think it'd be good to release within the next week. I'm happy to push on PRs, if any of you have time to review.

See [Milestone 0.19](https://github.com/python-versioneer/python-versioneer/milestone/6) for remaining tasks. Most will be taken care of by #219.

**Comment by astrojuanlu** [comment]
> My inclination is to drop Python < 3.6 before even trying to get the CI back in working order...

Not sure what's the status of this, but 3.6 will still receive security fixes for another 12 months and it would be great to have versioneer support a wide range of Python versions:

https://www.python.org/dev/peps/pep-0494/#and-beyond-schedule

Of course, that's up to the maintainers :) (also given that [projects like NumPy have moved past Python 3.6 already](https://numpy.org/neps/nep-0029-deprecation_policy.html))

Thanks a lot everyone for rescuing this wonderful package :heart:

**Comment by effigies** [comment]
We still support 3.6. We dropped 3.5 in the last release.

**Comment by sappelhoff** [comment]
It's great that maintenance of versioneer now continues as a group effort :rocket: I suggest to "pin"  this issue (and maybe give it some special tag) so that people can more easily find the discussion thread that lead to a "revival" of the package.

**Comment by effigies** [comment]
Just a heads up that I'm stepping down as a maintainer of this package (#359).

**Comment by Callek** [comment]
@effigies - fwiw - I don't anticipate being VERY active here, but I'll mention that I'm happy to take on ownership - at least in a "maintenance" stage.  (pypi releases, potential sec fixes, bugfixes, new python-version-support, etc)

The org I'm at has been using this for one of its current internal-only projects.   I should note that I additionally have experience with Mercurial from my time at Mozilla, so if there is interest in other VCS's I have enough experience to help shepherd them into this project.

**Comment by CallekTest** [comment]
> @Callek Sure. Searching around, you seem like a real person. I've sent you an invite as an outside collaborator for now. If it works for you, ping me on things you want to merge and I can give my feedback.

Awesome thanks, and yea I totally understand taking the cautious route here.  I am `Callek` on pypi if/when you want to add me there.

> > I should note that I additionally have experience with Mercurial from my time at Mozilla, so if there is interest in other VCS's I have enough experience to help shepherd them into this project.
> 
> Cool. My impression is that pretty much everybody just wants this to keep working as setuptools moves the ground from underneath them, and that the vendored code should play nicely with their linters/autoformatters/type checkers, rather than add the uncertainty of moving to an alternative like setuptools_scm. Given that setuptools_scm already supports Hg, I don't foresee many new users of versioneer from that direction.

Completely agreed here, just noting in case there is a future desire to support other VCS's. Continuing to support git(hub) and supporting setuptools changes, as well as following common linting desires for the vendored use indeed seems to be the current pain points!
